ext/temporary/CRM/Temporary/Meta.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+use CRM_Temporary_ExtensionUtil as E;
+
+class CRM_Temporary_Options {
+
+  public static function getTimestampModes(): array {
+    return [
+      'auto' => ts('(Auto)'),
+      'hybrid' => ts('Hybrid'),
+      'ts' => ts('TIMESTAMP Only'),
+      'gmt' => ts('DATETIME GMT Only'),
```

Comment:
The description may not have made this clear enough. The above setting is central to what the PR is saying. Basically:

* Take the whole data model and all the TIMESTAMP columns. (Or for smaller scope, set `TEMPORARY_TIMESTAMP_TABLES` and focus on specific tables.)
* For all these tables/fields, you can configure a general mode:
    * `TIMESTAMP Only`: This is what we have today.
    * `Hybrid`: Define both `TIMESTAMP` and `DATETIME-GMT` columns. Mirror data between them.
    * `DATETIME GMT Only`: Get rid of the TIMESTAMP columns

This gives a few knobs that can be dialed-in during a transition-plan.

For example, one transition plan might be "Everything in 9 months":

* __November__: Add extension as optional. Set `TEMPORARY_TIMESTAMP_TABLES` to all tables. Encourage more advanced/engaged folks to enable the extension with `Hybrid` mode. See if any trouble shakes-out.
* __January__: Make extension mandatory. Enable `Hybrid` mode on all sites. But you can still opt-out for a while.
* __February - July__: Old fields are deprecated. Start transitioning all the screens/reports/etc.
* __August__: Start running test-suites in mode `DATETIME GMT Only`. Deprecate `hybrid` and `ts` modes.
* __Next Major Version__: Drop support for `hybrid` and `ts` modes.

I'm not saying that ^^^ is a good or complete plan. There are issues. (Like: On large DBs, upgrades and reconfigurations would create lots of DB work.) A better plan might have some more staging (e.g. table-by-table; or batches of related tables). Or maybe all-tables-at-same-time is good. I'm just trying to illustrate how these are building-blocks for a transition plan.